typhon/objects/signals.py

```diff
@@ -0,0 +1,35 @@
+from rpython.rlib.rarithmetic import intmask
+
+from typhon.atoms import getAtom
+from typhon.autohelp import autohelp, method
+from typhon.objects.root import Object
+from typhon.objects.collections.maps import EMPTY_MAP
+from typhon.objects.data import IntObject
+
+from typhon.ruv import (alloc_signal, free, SignalStart, SignalStop,
+                        stashSignal, unstashSignal, unstashingSignal)
+
+RUN_1 = getAtom(u"run", 1)
+
+
+def _signalCB(signal, signum):
+    with unstashingSignal(signal) as (vat, handle):
+        vat.sendOnly(handle._target, RUN_1, [IntObject(intmask(signum))],
+                     EMPTY_MAP)
+
+
+@autohelp
+class SignalHandle(Object):
```

Comment:
This code seems too simple for the task placed upon it, TBH. I'm not sure what's missing, though.